internal/entitlement/postgresadapter/ent/schema/entitlement.go

```diff
@@ -30,6 +30,7 @@ func (Entitlement) Fields() []ent.Field {
 		field.String("feature_id").Immutable().SchemaType(map[string]string{
 			dialect.Postgres: "char(26)",
 		}),
+		field.String("feature_key").Immutable(),
```

Comment:
I don't think it's worth storing it here. I would rather prefer that the connector resolves the ID and we persits that data.

I wouldn't want to allow for a situation where a feature gets archived, then a new feature is created with the same key. Let's make sure that inside the DB we are not representing this convenience feature.

Reply:
So actually the problem has deeper roots than implementation details:

- If we want to allow entitlements to be accessible by featureKey, then it's necessary to limit that a user can only be entitled to a given featureKey once. Otherwise we cannot return a simple value.
- If we limit that then after archiving a feature the user can only be entitled again if the previous is deleted.

The issue is that this is not yet enforced (which i'll add) but after that these archivation scenarios are resolved. (this has some implications on the checks we do at create-time but nothing too complex)